src/tls/rsa_key_pair.h

```diff
@@ -213,6 +213,7 @@ namespace tls
         EVP_PKEY_encrypt(ctx, output.data(), &olen, input, input_size));
 
       output.resize(olen);
+      EVP_PKEY_CTX_free(ctx);
```

Comment:
Unless I'm missing something, OPENSSL_CHECK1() can throw, so we're still leaking in all error cases. Can we wrap the ctx into a unique_ptr with a custom deleter instead?

Reply:
Sure, can do. Failures of `OPENSSL_CHECK1` are pretty much fatal, so I didn't spend too much effort on cleanup, but I'll have another look around.

Reply:
We have a big catch around transaction logic though, in case anything gets thrown by user code. So these will get caught there, they won’t cause an abort. I’m happy for that to go in a follow up PR, but I would prefer if it was fixed.